### PR TITLE
fix: View resets to default when adding/removing a layer. #1690

### PR DESCRIPTION
**Pull request by nilscb**
- for some reason this has gone missing from the code at some point.

### CODE REVIEW

File: typescript/packages/subsurface-viewer/src/components/Map.tsx [hunk 1]
```diff
@@ -647,6 +654,9 @@ const Map: React.FC<MapProps> = ({
     }, [triggerHome]);
 
     useEffect(() => {
+        if (didUserChangeCamera && !isCameraPositionDefined) {
+            return;
+        }
         const [Views, viewStates] = createViewsAndViewStates(
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/components/Map.tsx:660]: Is what you want to achieve that `createViewsAndViewStates` is only called on first render and if camera is defined?

Comment (nilscb) [reply typescript/packages/subsurface-viewer/src/components/Map.tsx:660]: The aim is to not call "createViewsAndViewStates  (and hence reset the camera) if the user has manipulated the camera by interacting with it.

Comment (hkfb) [reply typescript/packages/subsurface-viewer/src/components/Map.tsx:660]: I think `onViewStateChanged` could be called for other reasons, like eg. syncing cameras across components, so maybe `didUserChangeCamera` should be named something like `viewStateChanged`?

**Review by hkfb — APPROVED** [review]

**Pull request merged**